projects/storefrontlib/src/cms-components/checkout/cart/add-to-cart/add-to-cart.component.ts

```diff
@@ -12,42 +13,73 @@ import { Observable } from 'rxjs';
 import { CartService, OrderEntry } from '@spartacus/core';
 
 import { AddedToCartDialogComponent } from './added-to-cart-dialog/added-to-cart-dialog.component';
+import { CurrentProductService } from '../../../product/current-product.service';
+import { filter } from 'rxjs/operators';
 
 @Component({
   selector: 'cx-add-to-cart',
   templateUrl: './add-to-cart.component.html',
-  styleUrls: ['./add-to-cart.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class AddToCartComponent implements OnInit {
-  modalInstance: any;
-
-  @Input()
-  iconOnly;
+  @Input() productCode: string;
 
-  @Input()
-  productCode: string;
-  @Input()
-  quantity: number;
+  @Input() showQuantity = true;
 
-  @Input()
   maxQuantity: number;
+  modalInstance: any;
+
+  hasStock = false;
+  quantity = 1;
 
   cartEntry$: Observable<OrderEntry>;
   loaded$: Observable<boolean>;
 
   constructor(
     protected cartService: CartService,
-    private modalService: NgbModal
+    private modalService: NgbModal,
+    protected currentProductService: CurrentProductService,
+    private cd: ChangeDetectorRef
   ) {}
 
   ngOnInit() {
     if (this.productCode) {
       this.loaded$ = this.cartService.getLoaded();
       this.cartEntry$ = this.cartService.getEntry(this.productCode);
+      this.hasStock = true;
+    } else {
+      this.currentProductService
+        .getProduct()
+        .pipe(filter(Boolean))
+        .subscribe(product => {
+          this.productCode = product.code;
+          if (
+            product.stock &&
+            product.stock.stockLevelStatus !== 'outOfStock'
+          ) {
+            this.maxQuantity = product.stock.stockLevel;
+            if (
+              product.stock.stockLevel > 0 ||
+              product.stock.stockLevelStatus === 'inStock'
+            ) {
+              this.hasStock = true;
```

Comment:
I don't think we need this because the first if statements already says that the stock is not out of stock, therefore I don't think we need to introduce a boolean, where we can just compare them in the template